plugin/runningCommand.go

```diff
@@ -0,0 +1,157 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+
+	"sync"
+)
+
+
+// RunningCommand represents a running command that was invoked by Execable#exec.
+// Use plugin.NewRunningCommand to create these objects.
+type RunningCommand struct {
+	ctx           context.Context
+	outputCh      chan ExecOutputChunk
+	stdout        *OutputStream
+	stderr        *OutputStream
+	mux           sync.Mutex
+	streamsClosed bool
+	exitCodeCh    chan int
+	exitCodeErrCh chan error
+}
+
+// NewRunningCommand creates a new RunningCommand object that's tied to
+// the passed-in execution context. When the execution context is cancelled,
+// the returned command's stdout/stderr streams will automatically be closed.
+// This means that you do not have to worry about handling dangling output
```

Comment:
Handling the cleanup here happened to fix a [bug in ssh.go](https://github.com/puppetlabs/wash/blob/master/transport/ssh.go#L144-L153) where the output streams were not closed prior to exiting the function.

Reply:
Oh, nevermind. According to https://stackoverflow.com/questions/8593645/is-it-ok-to-leave-a-channel-open, channels can be left open. I'll update the PR to simplify some of the cleanup code.